lib/after_commit_action.rb

```diff
@@ -17,6 +17,11 @@ module AfterCommitAction
   end
 
   def execute_after_commit(&block)
+    if ActiveRecord::Base.connection.open_transactions == 0 ||
+      ActiveRecord::Base.connection.add_transaction_record(self).nil?
+      return block.call
+    end
```

Comment:
that is actually what my code does, I'm just checking if it wasn't added to transaction, than I execute the block. Not sure why I added this check, don't remember if there were any problems with adding a record